### PR TITLE
Login redirects based on user role

### LINKED ISSUE #51: Login debe redirigir al /admin or /super-admin dependiendo del role

### PR DESCRIPTION
**Pull request by paulsancer**
close #51 

Also improved the way UserMenu logs out so the header can be aware and remove user email and menu.

### CODE REVIEW

**Comment by paulsancer** [comment]
@BernabeFelix there will be conflicts with your pr #56, I see you added withRoutes to `UserMenu.jsx` to redirect on `logout`, but I moved `logout` to `Header.jsx`...

File: client/src/components/common/Header/UserMenu.jsx [hunk 1]
```diff
@@ -44,10 +38,13 @@ class UserMenu extends Component {
           onRequestClose={this.closeMenu}
         >
           {isAdmin && (
-            <AdminMenu closeMenu={this.closeMenu} logout={this.logout} />
+            <AdminMenu closeMenu={this.closeMenu} logout={this.props.logout} />
```
Comment (BernabeFelix) [inline client/src/components/common/Header/UserMenu.jsx]: this is a ternary
isAdmin? AdminMenu: CustomerMenu ...

File: client/src/components/common/Header/Header.jsx [hunk 2]
```diff
@@ -46,7 +46,7 @@ class Header extends React.Component {
     const isAdmin = true;
     this.setState({
       userName: user && user.emailVerified ? user.email : null,
-      isAdmin
+      isAdmin: user && user.emailVerified ? isAdmin : null
```
Comment (BernabeFelix) [inline client/src/components/common/Header/Header.jsx]: why null and not false?

if false is an option, then `isAdmin: user && user.emailVerified`

**Review by BernabeFelix — CHANGES_REQUESTED** [review]

File: client/src/components/common/Header/Header.jsx (hunk 2 above)
Comment (paulsancer) [reply client/src/components/common/Header/Header.jsx]: Actually, I will better change it to `isAdmin` only, there is no point in validating that if `userName` decides whether the menu is shown or not.

File: client/src/components/common/Header/UserMenu.jsx (hunk 1 above)
Comment (paulsancer) [reply client/src/components/common/Header/UserMenu.jsx]: Duh... done :man_facepalming:

**Review by BernabeFelix — APPROVED** [review]

**Pull request merged**